app/pathogen/arbovirus/analyze/recharts/use-download-visualization.tsx

```diff
@@ -0,0 +1,47 @@
+import { MutableRefObject, useRef, useState } from 'react';
+import domtoimage from "dom-to-image";
+import fileDownload from "js-file-download";
+import { VisualizationId } from '../../visualizations/visualizations';
+
+interface UseDownloadVisualizationInput {
+  visualizationId: VisualizationId;
+}
+
+interface UseDownloadVisualizationOutput {
+  ref: MutableRefObject<HTMLDivElement | null>;
+  downloadVisualization: (input: DownloadVisualizationInput) => void;
+}
+
+interface DownloadVisualizationInput {
+  elementIdsToIgnore: string[]
+}
+
+export const useDownloadVisualization = (input: UseDownloadVisualizationInput): UseDownloadVisualizationOutput => {
+  const ref = useRef<HTMLDivElement>(null);
+
+  const downloadVisualization = (functionInput: DownloadVisualizationInput) => {
+    if(!ref.current) {
+      throw Error("Unable to generate image for visualization.")
+    }
+
+    domtoimage
```

Comment:
This library turned out to have almost instant download times while `htmltocanvas` was taking like 4-5 seconds. `htmltocanvas` had a bunch of messy interactions with recharts and `recharts-to-png` assumes that the element that `ref` is attached to is the recharts graph so I would lose the header and any padding around the graph which I can't do.

Anyways, that's what you're looking at here. This was the best solution for this by far.